### ISSUE TITLE
Restrcition gets applied to all categories and products instead of selected ones

### ISSUE THREAD

**Issue by suyati-saxena** [Vinai/groupscatalog2#117]
I have 10 categories, out of which i need to hide all but 1 for a customer group. So i have selected Hide From Groups for the 9 categories and kept [Use Default] for the remaining one category. For default values, i have used Category Mode - Show categories by default, Hide Categories from [None].
Now when i login with the customer group, other categories are not accessible, but the 1 which i have kept as Show is also not accessible and is redirected to the target route.
Similar thing happens with products as well. All of them have got disabled for the customer group. I cannot access any of them.
Can you please let me know as soon as possible if i m doing something wrong or there is some issue with the extension.
Thanks.

**Comment by Vinai** [comment]
Thanks for your detailed report. Did you try reindexing the groupscatalog tables? Does that cause something to be logged into the var/log/exception.log file?

**Comment by suyati-saxena** [comment]
Yes, i reindexed the groupcatalog multiple times. and no entry into exception.log file.

**Comment by Vinai** [comment]
Can you post the result of the following SQL queries please?

```sql
SELECT group_id, COUNT(*) FROM groupscatalog_category_idx group by group_id;
```
and
```sql
SELECT * FROM core_config_data WHERE path LIKE 'netzarbeiter_groupscatalog2/%';
```

**Comment by suyati-saxena** [comment]
![firefox_screenshot_2015-05-06t10-17-14 313z](https://cloud.githubusercontent.com/assets/6937594/7491449/379944ae-f407-11e4-86f0-b9789a86d7b5.png)
![firefox_screenshot_2015-05-06t10-17-06 241z](https://cloud.githubusercontent.com/assets/6937594/7491450/37b851dc-f407-11e4-9454-720a664e8b76.png)

Here are the screenshots

**Comment by Vinai** [comment]
Thanks. Your configuration looks fine. I looks like you have some specific visibility settings for the customer group 2. It seems that you have limited access only for that one customer group. Can you please give me the result of the following query?

```sql
SELECT value_id, store_id, entity_id, value FROM catalog_category_entity_text WHERE attribute_id=(SELECT attribute_id FROM eav_attribute WHERE entity_type_id=(SELECT entity_type_id FROM eav_entity_type WHERE entity_type_code = 'catalog_category') AND attribute_code = 'groupscatalog2_groups');
```
The result will be too large for phpmyadmin. Maybe you can export the result into a text file and copy & paste it in here, or run the command in a different MySQL client.

**Comment by suyati-saxena** [comment]
i reinstalled the module and tried if it is working with just few of the categories.
but it is not working..

**Comment by suyati-saxena** [comment]
![firefox_screenshot_2015-05-06t13-25-06 253z](https://cloud.githubusercontent.com/assets/6937594/7493806/76a64196-f421-11e4-80f6-dead0566ef0d.png)
Thats the screenshot for the last query you asked for. Another point i wanted to add is, i am using a static block for calling my category through their urls. Will this thing hinder the module?
I really need this to get working. I hope you can help me out with this.
Thanks.

**Comment by Vinai** [comment]
Could you please also post a screenshot of the backend interface for a generally accessable category and for the category that is hidden?
Thanks.

**Comment by suyati-saxena** [comment]
![new category manage categories categories catalogue magento admin](https://cloud.githubusercontent.com/assets/6937594/7494596/3a172c4a-f426-11e4-902d-b94bdbe753f4.png)
![firefox_screenshot_2015-05-06t13-51-31 096z](https://cloud.githubusercontent.com/assets/6937594/7494627/6920b542-f426-11e4-8380-16b54f1cd1b9.png)
Here are the screenshots..
Thanks..

**Comment by suyati-saxena** [comment]
Hello, 
Are we reaching to some solution?
I got this in my system.log file:

2015-05-07T10:02:53+00:00 ERR (3): Notice: unserialize(): Unexpected end of serialized data  in app/code/core/Mage/Admin/Model/Resource/User.php on line 193
2015-05-07T10:02:53+00:00 ERR (3): Notice: unserialize(): Error at offset 7590 of 7653 bytes  in app/code/core/Mage/Admin/Model/Resource/User.php on line 193

Not sure if it is related to the issue..
waiting for a response..

**Comment by joolswills** [comment]
I believe we may have the same problem on our multi website configuration - although with products this time - if unrelated I can open a new ticket.

we setup 700 or so products on website1 that are set to show to certain groups, and hide from most. 

This then broke the products on website2 (which had the default value set to [DEFAULT] (which is do not hide in the website config)) as they became hidden

we fixed this by setting all those 700 products to the value of NONE in the settings in each product. So when setting the config for one website, despite the default being visible, it will override it for other websites.

**Comment by Vinai** [comment]
I'm sorry to say that I have no time to contribute to my open source projects until next year due to the current work load. You will either have to take on debugging the issues yourself, of be patient.
Sorry for the inconvenience.

**Comment by joolswills** [comment]
NP. Will look into it myself and see what I come up with.

**Comment by Vinai** [comment]
Thank you. If you find a solution, it would be great if you would post it here. Others would appreciate it a lot I'm sure.
